django_elasticsearch_dsl/signals.py

```diff
@@ -54,22 +66,42 @@ def handle_save(self, sender, instance, **kwargs):
         Given an individual model instance, update the object in the index.
         Update the related objects either.
         """
-        registry.update(instance)
-        registry.update_related(instance)
+        try:
+            registry.update(instance)
+            registry.update_related(instance)
+
+        except (ElasticsearchException, ElasticsearchDslException):
+            if not LOG_ERRORS:
+                raise
+            LOGGER.exception('Error during auto-sync save, continuing')
 
     def handle_pre_delete(self, sender, instance, **kwargs):
         """Handle removing of instance object from related models instance.
         We need to do this before the real delete otherwise the relation
         doesn't exists anymore and we can't get the related models instance.
         """
-        registry.delete_related(instance)
+        try:
+            registry.delete_related(instance)
+
+        except (ElasticsearchException, ElasticsearchDslException):
+            if not LOG_ERRORS:
+                raise
+            LOGGER.exception('Error during auto-sync pre-delete, continuing')
 
     def handle_delete(self, sender, instance, **kwargs):
         """Handle delete.
 
         Given an individual model instance, delete the object from index.
         """
-        registry.delete(instance, raise_on_error=False)
+        try:
+            registry.delete(instance)
+
+        except Exception:
```

Comment:
Whats the reason behind capturing all the `Exception`? I think you can catch only the needed exceptions like above.

Reply:
The reasoning is to preserve what the method did previously. It passed a flag: `raise_on_error=False`. I removed that (so errors are raised, so I can log them). Then I emulate what used to happen by stuffing `Exception`.

Reply:
I am agree with @safwanrahman, the `raise_on_error` is used in https://elasticsearch-py.readthedocs.io/en/master/helpers.html#elasticsearch.helpers.bulk and by looking to the code of this method i think we only need to catch  https://github.com/elastic/elasticsearch-py/blob/de8413fdbdc6a207401a8cdc8091bf1dda319dab/elasticsearch/helpers/__init__.py#L13